src/check.ts

```diff
@@ -13,11 +13,11 @@ import {
 } from './helpers';
 
 export async function checkIssues(context: ActionContext) {
-	const { client, config, repo } = context;
+	const { client, readOnlyClient, config, repo } = context;
 
 	const manager = new IssueManager(client, repo, config);
 	const extractor = new DependencyExtractor(repo, config.keywords);
-	const resolver = new DependencyResolver(client, context.issues, repo);
+	const resolver = new DependencyResolver(client, readOnlyClient, context.issues, repo);
```

Comment:
You can pass `readOnlyClient` client as the first argument instead of `client`. that way you don't have to make changes in the `DependencyResolver` logic.